mifospay/src/main/java/org/mifos/mobilewallet/mifospay/merchants/ui/MerchantsFragment.java

```diff
@@ -0,0 +1,164 @@
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import org.mifos.mobilewallet.core.data.fineract.entity.accounts.savings.SavingsWithAssociations;
+import org.mifos.mobilewallet.mifospay.R;
+import org.mifos.mobilewallet.mifospay.base.BaseActivity;
+import org.mifos.mobilewallet.mifospay.base.BaseFragment;
+import org.mifos.mobilewallet.mifospay.merchants.MerchantsContract;
+import org.mifos.mobilewallet.mifospay.merchants.adapter.MerchantsAdapter;
+import org.mifos.mobilewallet.mifospay.merchants.presenter.MerchantsPresenter;
+import org.mifos.mobilewallet.mifospay.utils.Constants;
+import org.mifos.mobilewallet.mifospay.utils.RecyclerItemClickListener;
+import org.mifos.mobilewallet.mifospay.utils.Toaster;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class MerchantsFragment extends BaseFragment implements MerchantsContract.MerchantsView {
+
+    @Inject
+    MerchantsPresenter mPresenter;
+    MerchantsContract.MerchantsPresenter mMerchantsPresenter;
+
+    @Inject
+    MerchantsAdapter mMerchantsAdapter;
+
+    @BindView(R.id.inc_state_view)
+    View vStateView;
+    @BindView(R.id.rv_merchants)
+    RecyclerView mRvMerchants;
+    @BindView(R.id.iv_empty_no_transaction_history)
+    ImageView ivTransactionsStateIcon;
+
+    @BindView(R.id.tv_empty_no_transaction_history_title)
+    TextView tvTransactionsStateTitle;
+
+    @BindView(R.id.merchant_fragment_layout)
+    View mMerchantFragmentLayout;
+
+    @BindView(R.id.tv_empty_no_transaction_history_subtitle)
+    TextView tvTransactionsStateSubtitle;
+
+    @BindView(R.id.pb_merchants)
+    ProgressBar mMerchantProgressBar;
+    private List<SavingsWithAssociations> merchantsList;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((BaseActivity) getActivity()).getActivityComponent().inject(this);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_merchants, container, false);
+        ButterKnife.bind(this, rootView);
+        mPresenter.attachView(this);
+        mMerchantsPresenter.fetchMerchants();
+        setupUi();
+        return rootView;
+    }
+
+    private void setupUi() {
+        setUpSwipeRefreshLayout();
+        setUpRecyclerView();
+    }
+
+    private void setUpRecyclerView() {
+
+        mRvMerchants.setLayoutManager(new LinearLayoutManager(getContext()));
+        mRvMerchants.setAdapter(mMerchantsAdapter);
+        mRvMerchants.addOnItemTouchListener(new RecyclerItemClickListener(getContext(),
```

Comment:
Are we using this?

Reply:
Initially i was told to make the functionality of adding and removing merchants, thats why I put that clickListner .
Should I remove it ? @luckyman20 (I guess we can keep it for future use)

Reply:
It should be removed as of now as it might confuse somebody who is just exploring the project and we expect someone who is working on this functionality to know about this.

Reply:
Okay sure, fixing it.